community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/slurmsync.py

```diff
@@ -289,21 +289,17 @@ def nodes_unknown():
         inst = lkp.instance(first)
         log.error(f"{first} state: {state}, instance status:{inst.status}")
 
-    update = dict.get(
-        {
-            NodeStatus.orphan: nodes_delete,
-            NodeStatus.power_down: nodes_power_down,
-            NodeStatus.preempted: lambda: (nodes_down(), nodes_restart()),
-            NodeStatus.restore: nodes_idle,
-            NodeStatus.resume: nodes_resume,
-            NodeStatus.terminated: nodes_down,
-            NodeStatus.unbacked: nodes_down,
-            NodeStatus.unchanged: lambda: None,
-            NodeStatus.unknown: nodes_unknown,
-        },
-        status,
-    )
-    update()
+    {
+        NodeStatus.orphan: nodes_delete,
+        NodeStatus.power_down: nodes_power_down,
+        NodeStatus.preempted: lambda: (nodes_down(), nodes_restart()),
+        NodeStatus.restore: nodes_idle,
+        NodeStatus.resume: nodes_resume,
+        NodeStatus.terminated: nodes_down,
+        NodeStatus.unbacked: nodes_down,
+        NodeStatus.unchanged: lambda: None,
+        NodeStatus.unknown: nodes_unknown,
+    }[status]()
```

Comment:
Just curious what this ```[status]()``` syntax does here

Reply:
Here we create an anonymous map `NodeStatus -> function`, 
Then we perform a lookup function attributed to given `status`, by doing`[status]`
Finally we call found function `()`.

You can think of this construction as `dict(...).get(status).__call__()`;

The primary difference between old and new code is usage of `[status]` instead of `get(status)`.
If `X` IS NOT present in dict, `get(X)` will return `None`, while `[X]` will raise a `KeyError(X)` exception.

The exception is more preferable since it's more explicit about what went wrong `KeyError: NodeStatus.potatus"`. 
While `get` would successfully return `None`, but calling it `()` would fail with cryptic:
`TypeError: 'NoneType' object is not callable` 


See https://docs.python.org/3/library/stdtypes.html#mapping-types-dict